### PR TITLE
feat(compose): Add EFS support for task workflows

### PR DESCRIPTION
**Pull request by bvtujo**
Picks up from #1056 and adds tests, additional safety check for launch type, and adds test helper function to task_test.go. 

Addresses #1043 

----
Enter `[N/A]` in the box, if an item is not applicable to your change.

**Testing**
- [x] Unit tests passed
- [x] Integration tests passed
- [x] Unit tests added for new functionality
- [n/a] Listed manual checks and their outputs in the comments ([example](https://github.com/aws/amazon-ecs-cli/pull/750#issuecomment-472623042))
- [x] Link to issue or PR for the integration tests:

**Documentation**
- [ ] Contacted our doc writer
- [ ] Updated our README
----

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by bvtujo** [comment]
Integ tests launched 12:20pm. 

Will update this space with output.

File: ecs-cli/modules/cli/compose/entity/service/service_test.go [hunk 1]
```diff
@@ -183,10 +183,10 @@ func TestCreateWithEFSFargate(t *testing.T) {
 	)
 }
 
-func ecsParamsWithFargateEFSVolume() *utils.ECSParams {
+func ecsParamsWithEFSVolume(launchType string) *utils.ECSParams {
 	return &utils.ECSParams{
 		TaskDefinition: utils.EcsTaskDef{
-			ExecutionRole: "arn:aws:iam::123456789012:role/fargate_role",
+			ExecutionRole: fmt.Sprintf("arn:aws:iam::123456789012:role/%s_role", launchType),
```
Comment (allisaurus) [inline ecs-cli/modules/cli/compose/entity/service/service_test.go:189]: are we validating the role name elsewhere in the tests? wondering why we need to set it with a var

File: ecs-cli/modules/cli/compose/entity/task/task.go [hunk 2]
```diff
@@ -382,6 +383,14 @@ func (t *Task) buildRunTaskInput(taskDefinition string, count int, overrides map
 		}
 	}
 
+	// TODO: revert to "LATEST" when latest refers to 1.4.0
+	if launchType == config.LaunchTypeFargate && ecsParams != nil && len(ecsParams.TaskDefinition.EFSVolumes) != 0 {
```
Comment (allisaurus) [inline ecs-cli/modules/cli/compose/entity/task/task.go:387]: nit: can we make the last condition `... > 0` to be consistent with the service check

File: ecs-cli/modules/cli/compose/entity/service/service_test.go (hunk 1 above)
Comment (bvtujo) [reply ecs-cli/modules/cli/compose/entity/service/service_test.go:189]: You’re right, this doesn’t need to be parameterised. I did this before I realized that service and task were separate packages and though I might be able to share this function between the two test files.

File: ecs-cli/modules/cli/compose/entity/task/task.go [hunk 3]
```diff
@@ -382,6 +383,14 @@ func (t *Task) buildRunTaskInput(taskDefinition string, count int, overrides map
 		}
 	}
 
+	// TODO: revert to "LATEST" when latest refers to 1.4.0
+	if launchType == config.LaunchTypeFargate && ecsParams != nil && len(ecsParams.TaskDefinition.EFSVolumes) > 0 {
+		log.Warnf("Detected an EFS Volume in task definition %s", taskDefinition)
+		log.Warn("Using Fargate platform version 1.4.0, which includes changes to the networking flows for VPC endpoint customers.")
+		log.Warn("Learn more: https://aws.amazon.com/blogs/containers/aws-fargate-launches-platform-version-1-4/")
+		runTaskInput.PlatformVersion = aws.String("1.4.0")
```
Comment (SoManyHs) [inline ecs-cli/modules/cli/compose/entity/task/task.go:391]: Nit: wondering if it's worth constantizing this?

File: ecs-cli/modules/cli/compose/entity/task/task_test.go [hunk 4]
```diff
@@ -386,3 +387,101 @@ func TestBuildRuntaskInputManagedTagsDisabled(t *testing.T) {
...
+		CLIContext: cliContext,
+		ECSParams:  ecsParamsWithEFSVolume(),
+		CommandConfig: &config.CommandConfig{
+			Cluster:    cluster,
+			LaunchType: launchType,
+		},
+	}
+
+	task := &Task{
+		ecsContext: context,
```
Comment (SoManyHs) [inline ecs-cli/modules/cli/compose/entity/task/task_test.go:412]: Given that the test assertion only checks the name of the task def (and nothing about the actual EFS data), do you think it'd be worth asserting that part is all good or just assume the unit tests on convert_task_definition.go are enough?

Comment (bvtujo) [reply ecs-cli/modules/cli/compose/entity/task/task_test.go:412]: This test checks the platform version in the RunTaskInput struct, which I think is sufficient here.

**Pull request merged**